File: frontend/lib/domains/inbound/address_dto_in.dart
class AddressDTOIn {
  String? city;
  String? street;
  String? state;
  int? postalCode;
  String? country;

  AddressDTOIn({
    this.city,
    this.street,
    this.state,
    this.postalCode,
    this.country,
  });

  // Convert JSON to AddressDTOIn object
  factory AddressDTOIn.fromJson(Map<String, dynamic> json) {
    return AddressDTOIn(
      city: json['city'],
      street: json['street'],
      state: json['state'],
      postalCode: json['postalCode'],
      country: json['country'],
    );
  }

  // Convert AddressDTOIn object to JSON
  Map<String, dynamic> toJson() {
    return {
      'city': city,
      'street': street,
      'state': state,
      'postalCode': postalCode,
      'country': country,
    };
  }
}

File: backend/src/main/java/com/etekhno/rentacar/domain/inbound/AddressDTOIn.java
package com.etekhno.rentacar.domain.inbound;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class AddressDTOIn {
    @NotNull
    @NotBlank
    String city;
    @NotNull
    @NotBlank
    String street;
    @NotNull
    @NotBlank
    String state;
    @NotNull
    int postalCode;
    @NotNull
    @NotBlank
    String country;
}
